File: j2k/testData/fileOrElement/protected/constructorProperty.java
package test;

public abstract class Base {
    protected Activity mActivity;

    public Base(Activity activity) {
        mActivity = activity;
    }
}

public class Activity {
    public void display() {}
}

public class Derived extends Base {
    public Derived(Activity activity) {
        super(activity);
    }

    private View usage = new View() {
        @Override
        void click() {
            mActivity.display();
        }
    }
}

abstract class View {
    abstract void click();
}

File: j2k/testData/fileOrElement/protected/constructorProperty.kt
package test

abstract class Base(protected var mActivity: Activity)

class Activity {
    fun display() {
    }
}

class Derived(activity: Activity) : Base(activity) {

    private val usage = object : View() {
        override fun click() {
            mActivity.display()
        }
    }
}

internal abstract class View {
    internal abstract fun click()
}
